eng/build.sh

```diff
@@ -116,23 +117,46 @@ while [[ $# > 0 ]]; do
         shift 2
       fi
       ;;
+
      -arch)
       if [ -z ${2+x} ]; then 
         echo "No architecture supplied. See help (--help) for supported architectures." 1>&2
         exit 1
       fi
-      arch=$2
+      passedArch="$(echo "$2" | awk '{print tolower($0)}')"
+      case "$passedArch" in
+        x64|x86|arm|armel|arm64|wasm)
+          arch=$passedArch
+          ;;
+        *)
+          echo "Unsupported target architecture '$2'."
+          echo "The allowed values are x86, x64, arm, armel, arm64, and wasm."
+          exit 1
+          ;;
+      esac
       shift 2
       ;;
+
      -configuration|-c)
-     if [ -z ${2+x} ]; then 
+      if [ -z ${2+x} ]; then 
         echo "No configuration supplied. See help (--help) for supported configurations." 1>&2
         exit 1
       fi
-      val="$(tr '[:lower:]' '[:upper:]' <<< ${2:0:1})${2:1}"
+      passedConfig="$(echo "$2" | awk '{print tolower($0)}')"
+      case "$passedConfig" in
+        debug|release|checked)
+          val="$(tr '[:lower:]' '[:upper:]' <<< ${passedConfig:0:1})${passedConfig:1}"
```

Comment:
A nit - you can avoid this double transformation (once using awk and then using tr) by just keeping the original command and then passing the `val` to case and checking for `Debug`, `Release` and `Checked`.

Reply:
Do you mean by converting all to lowercase and then the first one to uppercase in the same command? I ask this because the `tr` command in `val` changes the first letter to uppercase, but leaves the rest untouched. That's why I'm using `awk` beforehand to convert everything to lowercase beforehand, in order to allow input like "RELEASE" to be valid. I'm removing the casing restrictions on the Linux scripts, aside from fixing the problems.

Reply:
I am sorry, I've misread the `tr` command.